homeassistant/components/smarttub/sensor.py

```diff
@@ -101,3 +162,10 @@ def device_state_attributes(self):
             ATTR_LAST_UPDATED: state.last_updated.isoformat(),
             ATTR_MODE: state.mode.name.lower(),
         }
+
+    async def async_set_secondary_filtration(self, **kwargs):
+        """Update primary filtration settings."""
+        mode = smarttub.SpaSecondaryFiltrationCycle.SecondaryFiltrationMode[
+            kwargs.get(ATTR_MODE).upper()
```

Comment:
```suggestion
            kwargs[ATTR_MODE].upper()
```

This can't be missing since its required